src/Microsoft.AspNetCore.Dispatcher/EndpointMiddleware.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Dispatcher
+{
+    public class EndpointMiddleware
+    {
+        private RequestDelegate _next;
+
+        public EndpointMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var feature = context.Features.Get<IDispatcherFeature>();
+            if (feature.RequestDelegate == null)
+            {
+                await _next(context);
+            }
+            else
+            {
+                await feature.RequestDelegate(context);
```

Comment:
It might be a problem that the RequestDelegate can't get a handle on `next`.

/cc @rynowak

Reply:
I don't understand why that would be desirable. It's called an **end**point

Reply:
@rynowak Because it restricts the types of things you can build with this system. I can't handle /foo and pass through to the next middleware.